### PR TITLE
Reproducing MIMOIRIS

### PR DESCRIPTION
**Pull request by YoshikiMas**
## What?

This PR adds a multi-talker ASR recipe for wsj0_2mix_spatialized. I will upload the pretrained models and add their links to README soon.

## Why?

To reproduce the paper entitled [Exploring the Integration of Speech Separation and Recognition with Self-Supervised Learning Representation](https://arxiv.org/abs/2307.12231). The scores are slightly different as I updated ESPnet and s3pel to the latest ones.

### CODE REVIEW

File: egs2/wsj/asr1/README.md [hunk 1]
```diff
@@ -28,6 +28,29 @@
...
+|---|---|---|---|---|---|---|---|---|
+|decode_lm_lm_train_lm_transformer_en_char_valid.loss.ave_asr_model_valid.acc.ave/test_dev93|503|8234|97.7|2.1|0.3|0.2|2.6|28.0|
+|decode_lm_lm_train_lm_transformer_en_char_valid.loss.ave_asr_model_valid.acc.ave/test_eval92|333|5643|98.6|1.3|0.1|0.1|1.5|19.5|
+
+### CER
+
+|dataset|Snt|Wrd|Corr|Sub|Del|Ins|Err|S.Err|
+|---|---|---|---|---|---|---|---|---|
+|decode_lm_lm_train_lm_transformer_en_char_valid.loss.ave_asr_model_valid.acc.ave/test_dev93|503|48634|99.1|0.3|0.5|0.2|1.0|37.8|
+|decode_lm_lm_train_lm_transformer_en_char_valid.loss.ave_asr_model_valid.acc.ave/test_eval92|333|33341|99.4|0.2|0.3|0.1|0.7|33.0|
```
Comment (Emrys365) [inline egs2/wsj/asr1/README.md:53]: Can you add some comments to point to the configuration file used to obtain this result?

File: egs2/wsj0_2mix_spatialized/enh1/README.md [hunk 2]
```diff
@@ -1,5 +1,30 @@
...
+
+
+## enh_train_enh_tfgridnet_waspaa2023_raw
+
+config: ./conf/tuning/train_enh_tfgridnet_waspaa2023.yaml
+
+|dataset|STOI|SAR|SDR|SIR|SI_SNR|
+|---|---|---|---|---|---|
+|enhanced_tt_spatialized_anechoic_multich_min_16k|99.65|27.45|26.98|38.43|27.02|
+|enhanced_tt_spatialized_reverb_multich_min_16k|98.13|18.81|18.39|29.47|18.12|
```
Comment (Emrys365) [inline egs2/wsj0_2mix_spatialized/enh1/README.md:25]: Is the model trained on anechoic or reverberant data?

File: espnet2/enh/separator/tfgridnet_separator.py [hunk 3]
```diff
@@ -71,13 +71,16 @@ def __init__(
...
+        ref_channel=-1,
     ):
         super().__init__()
         self.n_srcs = n_srcs
         self.n_layers = n_layers
         self.n_imics = n_imics
         assert n_fft % 2 == 0
         n_freqs = n_fft // 2 + 1
+        if ref_channel >= 0:
+            self.ref_channel = ref_channel
```
Comment (Emrys365) [inline espnet2/enh/separator/tfgridnet_separator.py:83]: Better to also handle the case where `ref_channel<0`.

**Review by Emrys365 — APPROVED** [review]
LSTM! Thanks.

File: egs2/wsj0_2mix_spatialized/enh1/README.md (hunk 2 above)
Comment (YoshikiMas) [reply egs2/wsj0_2mix_spatialized/enh1/README.md:26]: The model was trained on multi conditions (both anechoic and reverberant data). `mode=multi` indicates this point

**Pull request merged**